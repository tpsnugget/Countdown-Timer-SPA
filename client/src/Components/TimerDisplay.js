import React, { useState } from 'react'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import PauseCircleOutlineIcon from '@material-ui/icons/PauseCircleOutline';

import PlayCircleOutlineIcon from '@material-ui/icons/PlayCircleOutline';

const TimerDisplay = props => {

    const { min, moreThanHalfwayThere, pause, sec } = props

    return (
        <Grid container justify='center' alignItems='center' style={{marginTop: '3rem'}} >
            <Grid item xs={12}>
                {moreThanHalfwayThere ?
                <Typography variant='h6' >
                    More than halfway there!
                </Typography>
                : null}
            </Grid>
            <Grid container alignItems='center'>
                <Grid item xs={3}>
                    <Typography variant='h1'data-testid='Timerdisplay-time' >
                        {min} : {sec}
                    </Typography>
                </Grid>
                <Grid item xs={9}>
                {pause ?
                    <Button>
                        <PlayCircleOutlineIcon style={{fontSize: '4rem'}} />
                    </Button>
                    :
                    <Button data-testid='button' >
                        <PauseCircleOutlineIcon style={{fontSize: '4rem'}} />
                    </Button>
                }
                </Grid>
            </Grid>
        </Grid>
    )
}

TimerDisplay.propTypes = {
    min: PropTypes.number,
    moreThanHalfwayThere: PropTypes.bool,
    pause: PropTypes.bool,
    sec: PropTypes.number
}

export default TimerDisplay
