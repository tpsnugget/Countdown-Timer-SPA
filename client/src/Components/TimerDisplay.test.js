import React from 'react'
import { render } from '@testing-library/react'
import TimerDisplay from './TimerDisplay'

test('<TimerDisplay />', () => {
    const { container, getByTestId } = render(<TimerDisplay />)

    // Take a snapshot
    expect(container.firstChild).toMatchSnapshot()

    // Get the initial time
    const initialTime = getByTestId('Timerdisplay-time')
    expect(initialTime.textContent).toBe(' : ')

    // Get the button
    const submitButton = getByTestId('button')
    expect(submitButton).toHaveProperty('type', 'button')

})