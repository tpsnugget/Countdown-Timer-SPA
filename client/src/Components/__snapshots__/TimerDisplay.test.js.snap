// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerDisplay /> 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  style="margin-top: 3rem;"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6"
    >
      More than halfway there!
    </h6>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
        data-testid="Timerdisplay-time"
      >
         : 
      </h1>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        data-testid="button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="font-size: 4rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16h2V8H9v8zm3-14C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm1-4h2V8h-2v8z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
